src/Controls/src/Core/Platform/GestureManager/GesturePlatformManager.iOS.cs

```diff
@@ -512,40 +513,34 @@ weakEventTracker.Target is GesturePlatformManager eventTracker &&
 
 			return result;
 		}
-
-		static bool ShouldRecognizeTapsTogether(UIGestureRecognizer gesture, UIGestureRecognizer other)
+		internal class MauiUITapGestureRecognizer: UITapGestureRecognizer
 		{
-			// If multiple tap gestures are potentially firing (because multiple tap gesture recognizers have been
-			// added to the MAUI Element), we want to allow them to fire simultaneously if they have the same number
-			// of taps and touches
-
-			var tap = gesture as UITapGestureRecognizer;
-			if (tap == null)
-			{
-				return false;
-			}
-
-			var otherTap = other as UITapGestureRecognizer;
-			if (otherTap == null)
+			public MauiUITapGestureRecognizer(Action<UITapGestureRecognizer> action):base(action)
 			{
-				return false;
-			}
 
-			if (!Equals(tap.View, otherTap.View))
-			{
-				return false;
 			}
 
-			if (tap.NumberOfTapsRequired != otherTap.NumberOfTapsRequired)
-			{
-				return false;
-			}
+		}
 
-			if (tap.NumberOfTouchesRequired != otherTap.NumberOfTouchesRequired)
+		static bool ShouldRecognizeTapsTogether(UIGestureRecognizer gesture, UIGestureRecognizer other)
+		{
+			// If multiple tap gestures are potentially firing (because multiple tap gesture recognizers have been
+			// added to the MAUI Element), we want to allow them to fire simultaneously if they have the same number
+			// of taps and touches
+			var tap = gesture as MauiUITapGestureRecognizer;
+			var otherTap = other as MauiUITapGestureRecognizer;
```

Comment:
We can reduce the nesting a fair bit by:

```suggestion
			if (gesture is not MauiUITapGestureRecognizer tap)
				return true;
			if (other is not MauiUITapGestureRecognizer otherTap)
				return true;
```

Reply:
This is causing `TapGestureRecognizerNotWorkingOnEntry` to fail.